sdk/core/Azure.Core/src/TelemetryDetails.cs

```diff
@@ -80,16 +87,80 @@ internal static string GenerateUserAgentString(Assembly clientAssembly, string?
             {
                 version = version.Substring(0, hashSeparator);
             }
-            var platformInformation = $"({RuntimeInformation.FrameworkDescription}; {RuntimeInformation.OSDescription})";
+            runtimeInformation ??= new RuntimeInformationWrapper();
+            var platformInformation = EscapeProductInformation($"({runtimeInformation.FrameworkDescription}; {runtimeInformation.OSDescription})");
 
-            return applicationId != null
+            var ua = applicationId != null
                 ? $"{applicationId} azsdk-net-{assemblyName}/{version} {platformInformation}"
                 : $"azsdk-net-{assemblyName}/{version} {platformInformation}";
+            return ua;
         }
 
         /// <summary>
         /// The properly formatted UserAgent string based on this <see cref="TelemetryDetails"/> instance.
         /// </summary>
         public override string ToString() => _userAgent;
+
+        /// <summary>
+        /// If the ProductInformation is not in the proper format, this escapes any ')' , '(' or '\' characters per https://www.rfc-editor.org/rfc/rfc7230#section-3.2.6
+        /// </summary>
+        /// <param name="productInfo">The ProductInfo portion of the UserAgent</param>
+        /// <returns></returns>
+        private static string EscapeProductInformation(string productInfo)
+        {
+            // If the string is already valid, we don't need to escape anything
+            bool success = false;
+            try
+            {
+                success = ProductInfoHeaderValue.TryParse(productInfo, out var _);
+            }
+            catch (Exception)
+            {
+                // Invalid values can throw in Framework due to https://github.com/dotnet/runtime/issues/28558
+                // Treat this as a failure to parse.
+            }
+            if (success)
+            {
+                return productInfo;
+            }
+
+            var sb = new StringBuilder(productInfo.Length + 2);
+            sb.Append('(');
+            // exclude the first and last characters, which are the enclosing parentheses
+            for (int i = 1; i < productInfo.Length - 1; i++)
+            {
+                char c = productInfo[i];
+                if (c == ')' || c == '(')
+                {
+                    sb.Append('\\');
+                }
+                // If we see a \, we don't need to escape it if it's followed by a '\', '(', or ')', because it is already escaped.
+                else if (c == '\\')
+                {
+                    if (i + 1 < (productInfo.Length - 1))
+                    {
+                        char next = productInfo[i + 1];
+                        if (next == '\\' || next == '(' || next == ')')
+                        {
+                            sb.Append(c);
```

Comment:
nit: for consistency.

```suggestion
                            sb.Append(`\\`);
```

Reply:
Yes, this is intentional